src/main/java/com/hierynomus/msfscc/fileinformation/FileInformationFactory.java

```diff
@@ -107,6 +107,25 @@ public void write(FileBasicInformation info, Buffer outputBuffer) {
         decoders.put(FileBasicInformation.class, basicCodec);
         encoders.put(FileBasicInformation.class, basicCodec);
 
+        FileInformation.Codec<FileDispositionInformation> dispositionCodec = new FileInformation.Codec<FileDispositionInformation>() {
+            @Override
+            public FileInformationClass getInformationClass() {
+                return FileInformationClass.FileDispositionInformation;
+            }
+
+            @Override
+            public FileDispositionInformation read(Buffer inputBuffer) throws Buffer.BufferException {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void write(FileDispositionInformation info, Buffer outputBuffer) {
+                outputBuffer.putRawBytes(FileInformationFactory.getFileDispositionInfo(true));
```

Comment:
No, using `false` you can also unmark it for deletion. See: http://download.microsoft.com/download/4/3/8/43889780-8d45-4b2e-9d3a-c696a890309f/File%20System%20Behavior%20Overview.pdf